### ISSUE TITLE
Connection line always comes from first handle if there are multiple handles in custom node

### ISSUE THREAD

**Issue by zt7584** [bcakmakoglu/vue-flow#26]
Hi Braks,

Thank you for creating this library for Vue! This library is awesome.

I recently found out a small issue when trying to work with custom node. I found that when I try to drag and connect from one handle of my custom node to another node, the connection line always got plotted from the first handle that I defined for my custom node. This behavior seems to happen in the demo/examples as well, so it should be straightforward to recreate.

I was trying to dive into the code a bit, I found that it may be because of the following logics of picking [0] if certain information is not available:
```
 const sourceHandle =
  store.connectionHandleId && store.connectionHandleType
    ? props.sourceNode.handleBounds[store.connectionHandleType]?.find((d: HandleElement) => d.id === store.connectionHandleId)
    : store.connectionHandleType && props.sourceNode.handleBounds[store.connectionHandleType ?? 'source']?.[0]
```

I will also try to dive into code, when you get a time, could you help take a look at this small issue? 

Thanks!

**Comment by bcakmakoglu** [comment]
Hi!
Thanks for noticing the issue - I will check it out asap. You are of course free to dive in and if you find a solution, please open a PR =) 
But I will look into it!

**Comment by zt7584** [comment]
Hi Braks, thanks for the quick response! I was trying to dive into the code of master, and I actually found that this is fixed in master branch, the v0.4.0-0.

I was able to recreate this with tag v0.3.1. I think the issue may be because connectionHandleId was not able to get updated and passed correctly to ConnectionLine. The v0.4.0-0 seems to make a lot of refactoring around this area. That's my findings as of now. Sorry I am a bit new to vue, please forgive me if I am speaking non-sense here. :) Will continue to investigate and play around with it.

Quick question, if this is the case, will this get fixed in the v0.3.x major version? or will I expect to use v0.4.x major version? Thanks a lot for your help in advance!

**Comment by bcakmakoglu** [comment]
Ah, I've been absent from this project for a bit as I've started a new job this year so there might be bugs that have already been fixed with the upcoming version.
As of now 0.3.x will be deprecated and 0.4.x will include new features/bug fixes.

The release is pretty much done, I'm just stuck at writing the docs properly as that takes a little bit of time, seeing as there's been big changes in the versions, though ultimately those changes should make a lot of things much easier =)

**Comment by zt7584** [comment]
Hi Braks, got it, thanks for the insights! Looking forward to the new release!

**Comment by bcakmakoglu** [comment]
This got auto-closed with merge - was unintended.
I will leave this open until 0.4.0 is actually released.

**Comment by bcakmakoglu** [comment]
Closed and fixed with #62.

Please open a new issue if the problem somehow persists on your end.
